x/gov/keeper/min_deposit.go

```diff
@@ -78,18 +76,33 @@ func (keeper Keeper) GetLastMinDeposit(ctx sdk.Context) (sdk.Coins, time.Time) {
 
 // GetMinDeposit returns the (dynamic) minimum deposit currently required for a proposal
 func (keeper Keeper) GetMinDeposit(ctx sdk.Context) sdk.Coins {
+	minDeposit, _ := keeper.GetLastMinDeposit(ctx)
+
+	if minDeposit.Empty() {
+		// ValidateBasic prevents an empty FloorValue
+		// (and thus an empty deposit), if LastMinDeposit is empty
+		// it means it was never set, so we return the floor value
+		return keeper.GetParams(ctx).MinDepositThrottler.GetFloorValue()
+	}
+
+	return minDeposit
+}
+
+// UpdateMinDeposit updates the minimum deposit required for a proposal
+func (keeper Keeper) UpdateMinDeposit(ctx sdk.Context, checkElapsedTime bool) {
 	params := keeper.GetParams(ctx)
-	minDepositFloor := sdk.Coins(params.MinDepositThrottler.FloorValue)
 	tick := params.MinDepositThrottler.UpdatePeriod
+	lastMinDeposit, lastMinDepositTime := keeper.GetLastMinDeposit(ctx)
+	if checkElapsedTime && ctx.BlockTime().Sub(lastMinDepositTime).Nanoseconds() < tick.Nanoseconds() {
+		return
+	}
```

Comment:
There is no additional case in the unit tests for this new condition, is it because it is already covered or is it a miss ?